permissions/helpers.py

```diff
@@ -0,0 +1,273 @@
+        user, reservation_unit.unit
+    ) or is_superuser(user)
+
+
+def can_handle_application(user: User, application: Application) -> bool:
+    permission = "can_handle_applications"
+    return has_service_sector_permission(
+        user, [application.application_period.service_sector], permission
+    ) or is_superuser(user)
+
+
+def can_manage_service_sectors_application_rounds(
+    user: User, service_sector: ServiceSector
+) -> bool:
+    permission = "can_manage_application_rounds"
+    return (
+        has_general_permission(user, permission)
+        or has_service_sector_permission(user, [service_sector], permission)
+        or is_superuser(user)
+    )
+
+
+def can_modify_application_round(
+    user: User, application_round: ApplicationRound
+) -> bool:
+    return can_manage_service_sectors_application_rounds(
+        user, application_round.service_sector
+    )
+
+
+def can_manage_service_sectors_applications(
+    user: User, service_sector: ServiceSector
+) -> bool:
+    permission = "can_handle_applications"
+    return (
+        has_service_sector_permission(user, [service_sector], permission)
+        or has_general_permission(user, permission)
+        or is_superuser(user)
+    )
+
+
+def can_modify_application(user: User, application: Application) -> bool:
+    return (
+        can_manage_service_sectors_applications(
+            user, application.application_round.service_sector
+        )
+        or application.user == user
+        or is_superuser(user)
+    )
+
+
+def get_service_sectors_where_can_view_applications(user: User) -> list:
+    permission = "can_handle_applications"
+    if has_general_permission(user, permission) or is_superuser(user):
+        return list(ServiceSector.objects.all())
+
+    return list(
+        map(
+            lambda role: role.service_sector,
+            user.service_sector_roles.filter(role__permissions__permission=permission),
+        )
+    )
+
+
+def get_units_where_can_view_reservations(user: User) -> list:
+    permission = "can_view_reservations"
+    if has_general_permission(user, permission) or is_superuser(user):
+        return list(Unit.objects.all())
+
+    return list(
+        map(
+            lambda role: role.unit,
+            user.unit_roles.filter(role__permissions__permission=permission),
+        )
+    )
+
+
+def get_service_sectors_where_can_view_reservations(user: User) -> list:
+    permission = "can_view_reservations"
+    if has_general_permission(user, permission) or is_superuser(user):
+        return list(ServiceSector.objects.all())
+    return list(
+        map(
+            lambda role: role.service_sector,
+            user.service_sector_roles.filter(role__permissions__permission=permission),
+        )
+    )
+
+
+def can_view_reservation(user: User, reservation: Reservation) -> bool:
+    permission = "can_view_reservations"
+    reservation_units = reservation.reservation_unit.all()
```

Comment:
Mostly duplicate code here and the next method, might be ok so the permissions are super independent.